### PR TITLE
Add IPC and RPC config

### PR DESCRIPTION
**Pull request by dnkolegov**
This PR:

- removes deprecated mir-validator
- adds RPC config
- propogates the RPC URL to Mir validator types

As a result it fixes https://github.com/consensus-shipyard/lotus/issues/104

### CODE REVIEW

File: api/api_full.go [hunk 1]
```diff
@@ -862,7 +862,7 @@ type FullNode interface {
 	RaftState(ctx context.Context) (*RaftStateData, error) //perm:read
 	RaftLeader(ctx context.Context) (peer.ID, error)       //perm:read
 
-	// IPC-specific methods //
+	// IPCAgent-specific methods //
```
Comment (adlrocha) [inline api/api_full.go:865]: This is not correct. This are IPC-specific methods to call in Lotus, nothing to do with IPC agent.
```suggestion
	// IPC-specific methods //
```

File: chain/consensus/mir/config.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+type BaseConfig struct {
+	ID            address.Address
+	DatastorePath string
+	// InitialCheckpoint from which to start the validator.
+	InitialCheckpoint *checkpoint.StableCheckpoint
+	// CheckpointRepo determines the path where Mir checkpoints
+	// will be (optionally) persisted.
+	CheckpointRepo string
+	// The name of the group of validators.
+	GroupName string
```
Comment (adlrocha) [inline chain/consensus/mir/config.go:22]: How is this value used?

File: chain/consensus/mir/config.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+	"github.com/filecoin-project/mir/pkg/checkpoint"
+
+	"github.com/filecoin-project/lotus/chain/ipcagent/rpc"
+)
+
+// ---
+
+type BaseConfig struct {
+	ID            address.Address
```
Comment (adlrocha) [inline chain/consensus/mir/config.go:14]: Would you mind adding a comment here on what this ID is?

File: cmd/eudico/mirvalidator/run.go [hunk 4]
```diff
@@ -175,11 +180,23 @@ var runCmd = &cli.Command{
...
+			segmentLength,
+			cctx.String("ipcagent-url"),
+		)
+
 		var mb membership.Reader
 		switch cctx.String("membership") {
 		case "file":
 			mf := filepath.Join(cctx.String("repo"), cctx.String("membership-file"))
 			mb = membership.NewFileMembership(mf)
+		case "onchain":
```
Comment (adlrocha) [inline cmd/eudico/mirvalidator/run.go:197]: This assumes that as part of the command you have to pass the type of membership and the corresponding config as flags, right?

**Review by adlrocha — APPROVED** [review]
A few comments but LGTM. We may need a bit of rework on how are errors or empty validator memberships with subnets, but I guess this is something we need to de-risk ASAP in tests.

File: chain/consensus/mir/config.go (hunk 2 above)
Comment (dnkolegov) [reply chain/consensus/mir/config.go:22]: It is used for debugging. In tests the groupname is test name

File: chain/consensus/mir/config.go (hunk 3 above)
Comment (dnkolegov) [reply chain/consensus/mir/config.go:14]: Fixed

File: cmd/eudico/mirvalidator/run.go (hunk 4 above)
Comment (dnkolegov) [reply cmd/eudico/mirvalidator/run.go:197]: No. If the membership type is on-chain then on-chain membership reader will be created and passed to the Mir function as an argument.

File: api/api_full.go (hunk 1 above)
Comment (dnkolegov) [reply api/api_full.go]: Fixed

**Review by adlrocha — APPROVED** [review]
LGTM, let's merge and test

**Pull request merged**